trunk/SUAVE/Components/Energy/Networks/Battery_Propeller.py

```diff
@@ -90,151 +98,263 @@ def evaluate_thrust(self,state):
+                P[eta>1.0] = P[eta>1.0]*eta[eta>1.0]
+                F[eta[:,0]>1.0,:] = F[eta[:,0]>1.0,:]*eta[eta[:,0]>1.0,:]
+    
+                # Run the motor for current
+                _ , etam =  motor.current(conditions)
+                
+                # Conditions specific to this instantation of motor and propellers
+                R                   = prop.tip_radius
+                rpm                 = motor.outputs.omega / Units.rpm
+                F_mag               = np.atleast_2d(np.linalg.norm(F, axis=1)).T
+                total_thrust        = total_thrust + F * factor
+                total_power         = total_power  + P * factor
+                total_motor_current = total_motor_current + factor*motor.outputs.current
+    
+                # Pack specific outputs
+                conditions.propulsion.propeller_motor_efficiency[:,ii] = etam[:,0]  
+                conditions.propulsion.propeller_motor_torque[:,ii]     = motor.outputs.torque[:,0]
+                conditions.propulsion.propeller_torque[:,ii]           = Q[:,0]
+                conditions.propulsion.propeller_rpm[:,ii]              = rpm[:,0]
+                conditions.propulsion.propeller_tip_mach[:,ii]         = (R*rpm[:,0]*Units.rpm)/a[:,0]
+                conditions.propulsion.disc_loading[:,ii]               = (F_mag[:,0])/(np.pi*(R**2)) # N/m^2                  
+                conditions.propulsion.power_loading[:,ii]              = (F_mag[:,0])/(P[:,0])      # N/W      
+                conditions.propulsion.propeller_efficiency[:,ii]       = etap[:,0]      
+                    
+                conditions.noise.sources.propellers[prop.tag]          = outputs
+    
+            # Run the avionics
+            avionics.power()
+    
+            # Run the payload
+            payload.power()
+            
+            # link
+            esc.inputs.currentout = total_motor_current
     
+            # Run the esc
+            esc.currentin(conditions)  
+            
+            # Calculate avionics and payload power
+            avionics_payload_power = avionics.outputs.power + payload.outputs.power
+        
+            # Calculate avionics and payload current
+            avionics_payload_current = avionics_payload_power/self.voltage 
+        
+            # link
+            battery.inputs.current  = esc.outputs.currentin + avionics_payload_current
+            battery.inputs.power_in = -(esc.outputs.voltageout *esc.outputs.currentin + avionics_payload_power)
+            battery.energy_cycle_model(numerics,discharge_flag)         
+             
+        # --------------------------------------------------------------------------------
+        # Run Charge Model 
+        # --------------------------------------------------------------------------------               
+        else:  
+            # link 
+            battery.inputs.current  = -battery.cell.charging_current*n_parallel * np.ones_like(volts)
```

Comment:
can we do this math somewhere else?

Reply:
these are just one-liners. I do not one to create unnecessary functions.